tools/cuds-prioritized-attribution/composer/main.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+import os
+from typing import List, Dict, Optional, Union
+from airflow import models
+from airflow.operators import python_operator
+from google.cloud import bigquery
+from dependencies import billingoutput
+from dependencies import commitment_intervals
+from dependencies import commitments_schema
+from dependencies import distribute_commitment
+from dependencies import helper_function
+from dependencies import project_label_credit_data
+
+
+def get_env_variables(key_list: List[str]) -> Dict[str, Union[Optional[str],
+                                                              bool]]:
+    """Creates a Dictionary object to hold all of the environment variables.
+
+    Args:
+        key_list: List of strings of environment variable keys.
+
+    Returns:
+        Dictionary holding key-value pairs of environment variables.
+    """
+    return {key: os.environ.get(key) for key in key_list}
+
+
+def execute_query(bq_client: bigquery.Client,
```

Comment:
You are not leveraging airflow's native functionality and this is prone to not working for a lot of people, and is a maintenance burden.
If you're not going to use airflow constructs this might as well be a cron job that invokes python scripts to execute bigquery queries (which we also have had customers do https://github.com/openx/ox-bqpipeline/).
Why are you not using BigQueryOperator? This would use BigQueryHook to manage the bigquery credentials, have improved logging and error handling etc.

I notice you are rendering parts of the sql string based on env vars but BigQueryOperator supports rendering jinja templates and bigquery query parameters.

Reply:
BQ Query Parameters can't be used for table names I believe, hence why we're using env vars.